### ISSUE TITLE
Ability to get values safely from arrays/dicts without errors

### ISSUE THREAD

**Issue by michidk** [typst/typst#946]
# Intrduction
The arrays and dicts `at` function can be used to access attributes by index/name. If the index is not found, it will return an error:
https://github.com/typst/typst/blob/main/src/eval/dict.rs#L57-L59

# Usecases
In most cases one would want to have some different value in case the index is not found. This is especially the case when working with external data sources (e.g. json files, with very flexible schemas). In those cases, the following pattern is often used:
```typst
#let value = if "something" in list {
    list.at(key)
  } else {
    default
  }
```

# Workaround
The current workaround that I have in all my typst codebases is a helper function that gets the item and returns a default value in case it's not in the keys of the list:
```
// safely get a value from a list, or return a default value
#let get(list, key, default) = {
  if key in list {
    list.at(key)
  } else {
    default
  }
}
```

# Implementation
As @laurmaedje pointed our [here](https://github.com/typst/typst/pull/945#issuecomment-1519053446), there are different ways of implementing this.

1. Modifying `at` to instead of throwing an error returning `none`
2. Modifying `at` to have an optional default argument `dict.at(key, default: none)` (which is an extension to 1.)
3. A new method `dict.get()` that returns `none` instead of erroring
4. A new method `dict.get(key, default)`: https://github.com/typst/typst/pull/945
5. A new method `dict.get(key, default: none)`, which is a mix of 3. and 4.

With every implementation where we return `none` (every on the suggestions above but 4.), we will have the issue that `none` will be the return value, if either the element does not exist or the value exists and was actually just `none`. Javascript solves this by having `null` and `undefined`. I think that this is probably not a big problem in practice, but feels like bad design. However, this makes it's use really neat: `dict.get("key") or "default"`.

In general, I would prefer 4. Though also 1. seems a nice solution, when we are okay with returning `none`.

**Comment by Marmare314** [comment]
One option which isn't listed here (and I like the most) is 2 without 1. Have `at` still throw an error unless a default is specified.

**Comment by laurmaedje** [comment]
> One option which isn't listed here (and I like the most) is 2 without 1. Have at still throw an error unless a default is specified.

I think I'd be on board with this. Should we have a more general solution in the future, we can still deprecate/remove it.

**Comment by michidk** [comment]
I agree; I implemented this approach in #955.

**Comment by laurmaedje** [comment]
Fixed via #995.
